composer/utils/checkpoint.py

```diff
@@ -36,7 +35,7 @@
 
 _COMPOSER_STATES_FILENAME = 'composer_states.pt'
 _DEEPSPEED_TAG = 'deepspeed'  # always tag with the same, deterministic name. We'll rename the tarball to the appropriate name.
-
+_TORCH_DISTRIBUTED_CHECKPOINTS_FILENAME = f"__{dist.get_global_rank()}_0.distcp"
```

Comment:
why is there a zero at the end of the filename?

Reply:
it's the file count per rank and we set `single_file_per_rank` to True, so it will always be zero
https://github.com/pytorch/pytorch/blob/9a2df0a5af010201f31c87c593227659ae1adc4e/torch/distributed/checkpoint/filesystem.py#L371-L383